src/fileio_type.h

```diff
@@ -40,6 +40,9 @@ enum DetailedFileType {
 	DFT_FIOS_DIR,    ///< A directory entry.
 	DFT_FIOS_DIRECT, ///< Direct filename.
 
+	/* Ending variable used for compile time check of this array against _fios_colours in fiose_gui.cpp */
+	DFT_END,
```

Comment:
This sort of specific comment seems like an invitation to be outdated someday. I'd suggest something more generic, formatted as an inline Doxygen comment.

```suggestion
	DFT_END, ///< End-of-array marker.
```